### ISSUE TITLE
glibc-riscv64-unknown-linux-gnu build failure

### ISSUE THREAD

**Issue by rowanG077** [NixOS/nixpkgs#130121]
**Describe the bug**
glibc-riscv64-unknown-linux-gnu fails to build with the following linker error:

```
...
/nix/store/d57ai5nz7rn4h179lq5bvkfcy8gmfnr6-riscv64-unknown-linux-gnu-binutils-2.35.1/bin/riscv64-unknown-linux-gnu-ld: /build/build/libc.a(dl-reloc-static-pie.o): in function `_dl_relocate_static_pie':
```

**To Reproduce**
1. import current cross packages:
```
riscCrossPkgs = import (fetchTarball {
    name = "nixpkgs-master-2021-07-13";
    url = "https://github.com/nixos/nixpkgs/archive/f082c435dd187f92b8423413dff9d0b11c7195f7.tar.gz";
    sha256 = "0mkq6isl96ls70593wqxkd9rx68zhi9qgr97jq1x88pvj2h68r7k";
  }) {
    crossSystem = {
      config = "riscv64-unknown-linux-gnu";
    };
  };
```
2. Use this pkgs and try to get glibc: `riscCrossPkgs.glibc`

**Notify maintainers**
I'm not sure who is the correct person to notify. @shlevy You seem to have done a bit of riscv stuff :). Sorry to bother you with this but please tag the appropriate person if you know who it is. 

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.8, NixOS, 21.05pre271938.c7d0dbe094c (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(rowan.goemans): `""`
 - channels(root): `"nixos-21.05pre279012.ad47284f8b0, nixos-hardware, stable-20.09.3654.3a8a85da229"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
Basically everything that depends on riscv cross compilation.

**Comment by piegamesde** [comment]
Failure log:

```
make[4]: Entering directory '/build/glibc-2.32/csu'
make subdir=dirent -C ../dirent ..=../ objdir=/build/build -f Makefile -f ../elf/rtld-Rules rtld-all rtld-modules='rtld-closedir.os rtld-rewinddir.os rtld-readdir64.os rtld-fdopendir.os rtld-rewinddir.os rtld-rewinddir.os rtld-getdents64.os rtld-readdir64.os'
make[4]: Leaving directory '/build/glibc-2.32/csu'
make subdir=elf -C ../elf ..=../ objdir=/build/build -f Makefile -f ../elf/rtld-Rules rtld-all rtld-modules='rtld-dl-addr-obj.os'
make[4]: Entering directory '/build/glibc-2.32/elf'
make[4]: Entering directory '/build/glibc-2.32/dirent'
riscv64-unknown-linux-gnu-gcc ldconfig.c -c -std=gnu11 -fgnu89-inline  -g -O2 -Wall -Wwrite-strings -Wundef -Werror -fmerge-all-constants -frounding-math -fno-stack-protector -Wstrict-prototypes -Wold-style-definition -fmath-errno   -fpie  -D'PREFIX="/nix/store/4j6n0giv0iqqqh1c5r5bpqmyw6n20qjl-glibc-riscv64-unknown-linux-gnu-2.32-48"' -D'LIBDIR="/nix/store/4j6n0giv0iqqqh1c5r5bpqmyw6n20qjl-glibc-riscv64-unknown-linux-gnu-2.32-48/lib"' -D'SLIBDIR="/nix/store/4j6n0giv0iqqqh1c5r5bpqmyw6n20qjl-glibc-riscv64-unknown-linux-gnu-2.32-48/lib"'        -I../include -I/build/build/elf  -I/build/build  -I../sysdeps/unix/sysv/linux/riscv/rv64  -I../sysdeps/unix/sysv/linux/riscv  -I../sysdeps/riscv/nptl  -I../sysdeps/unix/sysv/linux/generic  -I../sysdeps/unix/sysv/linux/wordsize-64  -I../sysdeps/unix/sysv/linux/include -I../sysdeps/unix/sysv/linux  -I../sysdeps/nptl  -I../sysdeps/pthread  -I../sysdeps/gnu  -I../sysdeps/unix/inet  -I../sysdeps/unix/sysv  -I../sysdeps/unix  -I../sysdeps/posix  -I../sysdeps/riscv/rv64/rvd  -I../sysdeps/riscv/rv64/rvf  -I../sysdeps/riscv/rvf  -I../sysdeps/riscv/rvd  -I../sysdeps/riscv/rv64  -I../sysdeps/riscv  -I../sysdeps/ieee754/ldbl-128  -I../sysdeps/ieee754/dbl-64  -I../sysdeps/ieee754/flt-32  -I../sysdeps/wordsize-64  -I../sysdeps/ieee754  -I../sysdeps/generic  -I.. -I../libio -I. -nostdinc -isystem /nix/store/sb9wd3n8g51ykzhz2zn7wrwrl9l2a4pn-riscv64-unknown-linux-gnu-stage-static-gcc-debug-10.3.0/lib/gcc/riscv64-unknown-linux-gnu/10.3.0/include -isystem /nix/store/sb9wd3n8g51ykzhz2zn7wrwrl9l2a4pn-riscv64-unknown-linux-gnu-stage-static-gcc-debug-10.3.0/lib/gcc/riscv64-unknown-linux-gnu/10.3.0/include-fixed -isystem /nix/store/5l5ljdpa939rg2jrk122y5sfb3sf0r7w-linux-headers-5.12/include -D_LIBC_REENTRANT -include /build/build/libc-modules.h -DMODULE_NAME=ldconfig -include ../include/libc-symbols.h  -DPIC     -DTOP_NAMESPACE=glibc -o /build/build/elf/ldconfig.o -MD -MP -MF /build/build/elf/ldconfig.o.dt -MT /build/build/elf/ldconfig.o
make[4]: Leaving directory '/build/glibc-2.32/dirent'
make subdir=gmon -C ../gmon ..=../ objdir=/build/build -f Makefile -f ../elf/rtld-Rules rtld-all rtld-modules='rtld-profil.os rtld-prof-freq.os rtld-profil.os rtld-profil.os rtld-profil.os'
make[4]: Entering directory '/build/glibc-2.32/gmon'
make[4]: Leaving directory '/build/glibc-2.32/gmon'
make subdir=io -C ../io ..=../ objdir=/build/build -f Makefile -f ../elf/rtld-Rules rtld-all rtld-modules='rtld-xstat.os rtld-fxstat.os rtld-lxstat.os rtld-lseek64.os rtld-access.os rtld-close_nocancel.os rtld-fcntl_nocancel.os rtld-open64_nocancel.os rtld-openat64_nocancel.os rtld-read_nocancel.os rtld-pread64_nocancel.os rtld-write_nocancel.os'
make[4]: Entering directory '/build/glibc-2.32/io'
riscv64-unknown-linux-gnu-gcc pldd.c -c -std=gnu11 -fgnu89-inline  -g -O2 -Wall -Wwrite-strings -Wundef -Werror -fmerge-all-constants -frounding-math -fno-stack-protector -Wstrict-prototypes -Wold-style-definition -fmath-errno   -fpie          -I../include -I/build/build/elf  -I/build/build  -I../sysdeps/unix/sysv/linux/riscv/rv64  -I../sysdeps/unix/sysv/linux/riscv  -I../sysdeps/riscv/nptl  -I../sysdeps/unix/sysv/linux/generic  -I../sysdeps/unix/sysv/linux/wordsize-64  -I../sysdeps/unix/sysv/linux/include -I../sysdeps/unix/sysv/linux  -I../sysdeps/nptl  -I../sysdeps/pthread  -I../sysdeps/gnu  -I../sysdeps/unix/inet  -I../sysdeps/unix/sysv  -I../sysdeps/unix  -I../sysdeps/posix  -I../sysdeps/riscv/rv64/rvd  -I../sysdeps/riscv/rv64/rvf  -I../sysdeps/riscv/rvf  -I../sysdeps/riscv/rvd  -I../sysdeps/riscv/rv64  -I../sysdeps/riscv  -I../sysdeps/ieee754/ldbl-128  -I../sysdeps/ieee754/dbl-64  -I../sysdeps/ieee754/flt-32  -I../sysdeps/wordsize-64  -I../sysdeps/ieee754  -I../sysdeps/generic  -I.. -I../libio -I. -nostdinc -isystem /nix/store/sb9wd3n8g51ykzhz2zn7wrwrl9l2a4pn-riscv64-unknown-linux-gnu-stage-static-gcc-debug-10.3.0/lib/gcc/riscv64-unknown-linux-gnu/10.3.0/include -isystem /nix/store/sb9wd3n8g51ykzhz2zn7wrwrl9l2a4pn-riscv64-unknown-linux-gnu-stage-static-gcc-debug-10.3.0/lib/gcc/riscv64-unknown-linux-gnu/10.3.0/include-fixed -isystem /nix/store/5l5ljdpa939rg2jrk122y5sfb3sf0r7w-linux-headers-5.12/include -D_LIBC_REENTRANT -include /build/build/libc-modules.h -DMODULE_NAME=nonlib -include ../include/libc-symbols.h  -DPIC     -DTOP_NAMESPACE=glibc -o /build/build/elf/pldd.o -MD -MP -MF /build/build/elf/pldd.o.dt -MT /build/build/elf/pldd.o
riscv64-unknown-linux-gnu-gcc -nostdlib -nostartfiles -static -o /build/build/elf/sln -static-pie    /build/build/csu/rcrt1.o /build/build/csu/crti.o `riscv64-unknown-linux-gnu-gcc  --print-file-name=crtbeginS.o` /build/build/elf/sln.o /build/build/elf/static-stubs.o  -Wl,-z,now -Wl,--start-group /build/build/libc.a -lgcc  -Wl,--end-group `riscv64-unknown-linux-gnu-gcc  --print-file-name=crtendS.o` /build/build/csu/crtn.o
make[4]: Leaving directory '/build/glibc-2.32/io'
make subdir=malloc -C ../malloc ..=../ objdir=/build/build -f Makefile -f ../elf/rtld-Rules rtld-all rtld-modules='rtld-scratch_buffer_set_array_size.os'
make[4]: Entering directory '/build/glibc-2.32/malloc'
make[4]: Leaving directory '/build/glibc-2.32/malloc'
make subdir=misc -C ../misc ..=../ objdir=/build/build -f Makefile -f ../elf/rtld-Rules rtld-all rtld-modules='rtld-mmap64.os rtld-munmap.os rtld-mprotect.os'
riscv64-unknown-linux-gnu-gcc -nostdlib -nostartfiles -static -o /build/build/elf/ldconfig -static-pie    /build/build/csu/rcrt1.o /build/build/csu/crti.o `riscv64-unknown-linux-gnu-gcc  --print-file-name=crtbeginS.o` /build/build/elf/ldconfig.o /build/build/elf/cache.o /build/build/elf/readlib.o /build/build/elf/xmalloc.o /build/build/elf/xstrdup.o /build/build/elf/chroot_canon.o /build/build/elf/static-stubs.o  -Wl,-z,now -Wl,--start-group /build/build/libc.a -lgcc  -Wl,--end-group `riscv64-unknown-linux-gnu-gcc  --print-file-name=crtendS.o` /build/build/csu/crtn.o
/nix/store/d57ai5nz7rn4h179lq5bvkfcy8gmfnr6-riscv64-unknown-linux-gnu-binutils-2.35.1/bin/riscv64-unknown-linux-gnu-ld: /build/build/libc.a(dl-reloc-static-pie.o): in function `_dl_relocate_static_pie':
/build/glibc-2.32/elf/dl-reloc-static-pie.c:29: undefined reference to `_DYNAMIC'
collect2: error: ld returned 1 exit status
make[2]: *** [../Rules:256: /build/build/elf/sln] Error 1
make[2]: *** Waiting for unfinished jobs....
make subdir=nptl -C ../nptl ..=../ objdir=/build/build -f Makefile -f ../elf/rtld-Rules rtld-all rtld-modules='rtld-libc-lowlevellock.os'
make[4]: Entering directory '/build/glibc-2.32/misc'
/nix/store/d57ai5nz7rn4h179lq5bvkfcy8gmfnr6-riscv64-unknown-linux-gnu-binutils-2.35.1/bin/riscv64-unknown-linux-gnu-ld: /build/build/libc.a(dl-reloc-static-pie.o): in function `_dl_relocate_static_pie':
/build/glibc-2.32/elf/dl-reloc-static-pie.c:29: undefined reference to `_DYNAMIC'
collect2: error: ld returned 1 exit status
make[2]: *** [../Rules:256: /build/build/elf/ldconfig] Error 1
make subdir=posix -C ../posix ..=../ objdir=/build/build -f Makefile -f ../elf/rtld-Rules rtld-all rtld-modules='rtld-uname.os rtld-_exit.os rtld-getpid.os rtld-environ.os rtld-uname.os'
make[4]: Leaving directory '/build/glibc-2.32/misc'
make subdir=setjmp -C ../setjmp ..=../ objdir=/build/build -f Makefile -f ../elf/rtld-Rules rtld-all rtld-modules='rtld-setjmp.os rtld-__longjmp.os'
make[4]: Entering directory '/build/glibc-2.32/setjmp'
make[4]: Leaving directory '/build/glibc-2.32/elf'
make subdir=signal -C ../signal ..=../ objdir=/build/build -f Makefile -f ../elf/rtld-Rules rtld-all rtld-modules='rtld-sigaction.os'
make[4]: Leaving directory '/build/glibc-2.32/setjmp'
make subdir=string -C ../string ..=../ objdir=/build/build -f Makefile -f ../elf/rtld-Rules rtld-all rtld-modules='rtld-strchr.os rtld-strcmp.os rtld-strcspn.os rtld-strdup.os rtld-strlen.os rtld-strdup.os rtld-strnlen.os rtld-strncmp.os rtld-memchr.os rtld-memcmp.os rtld-memmove.os rtld-memset.os rtld-stpcpy.os rtld-memcpy.os rtld-wordcopy.os rtld-memmove.os rtld-rawmemchr.os rtld-strchrnul.os rtld-strcspn.os'
make[4]: Entering directory '/build/glibc-2.32/posix'
make[4]: Entering directory '/build/glibc-2.32/signal'
make[4]: Entering directory '/build/glibc-2.32/string'
../Rules:307: warning: overriding recipe for target '/build/build/string/tst-strsignal.out'
../Rules:294: warning: ignoring old recipe for target '/build/build/string/tst-strsignal.out'
../Rules:307: warning: overriding recipe for target '/build/build/string/tst-strerror.out'
../Rules:294: warning: ignoring old recipe for target '/build/build/string/tst-strerror.out'
make[4]: Leaving directory '/build/glibc-2.32/signal'
make subdir=time -C ../time ..=../ objdir=/build/build -f Makefile -f ../elf/rtld-Rules rtld-all rtld-modules='rtld-setitimer.os'
make[4]: Entering directory '/build/glibc-2.32/time'
make[4]: Leaving directory '/build/glibc-2.32/posix'
../Rules:307: warning: overriding recipe for target '/build/build/string/tst-strsignal.out'
../Rules:294: warning: ignoring old recipe for target '/build/build/string/tst-strsignal.out'
../Rules:307: warning: overriding recipe for target '/build/build/string/tst-strerror.out'
../Rules:294: warning: ignoring old recipe for target '/build/build/string/tst-strerror.out'
make[4]: Leaving directory '/build/glibc-2.32/string'
make[4]: Entering directory '/build/glibc-2.32/nptl'
make[4]: Leaving directory '/build/glibc-2.32/time'
make[4]: Leaving directory '/build/glibc-2.32/nptl'
make[3]: Leaving directory '/build/glibc-2.32/elf'
make[2]: Leaving directory '/build/glibc-2.32/elf'
make[1]: *** [Makefile:470: elf/others] Error 2
make[1]: Leaving directory '/build/glibc-2.32'
make: *** [Makefile:9: all] Error 2
builder for '/nix/store/w0jckdmvk7qym4d8yj089jsj8bjv1imp-glibc-riscv64-unknown-linux-gnu-2.32-48.drv' failed with exit code 2
error: build of '/nix/store/w0jckdmvk7qym4d8yj089jsj8bjv1imp-glibc-riscv64-unknown-linux-gnu-2.32-48.drv' on 'ssh://builder1' failed: builder for '/nix/store/w0jckdmvk7qym4d8yj089jsj8bjv1imp-glibc-riscv64-unknown-linux-gnu-2.32-48.drv' failed with exit code 2
builder for '/nix/store/w0jckdmvk7qym4d8yj089jsj8bjv1imp-glibc-riscv64-unknown-linux-gnu-2.32-48.drv' failed with exit code 1
cannot build derivation '/nix/store/gg1ybmvl8gzc3jpy6gv0qdq2xh49nlda-riscv64-unknown-linux-gnu-binutils-wrapper-2.35.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/ly0c8x6zwdmhf5ip7w3277jkbsp7hlq5-riscv64-unknown-linux-gnu-stage-final-gcc-debug-10.3.0.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/f3qfckhryf4mnx9i0s7fca4bs221mlqa-riscv64-unknown-linux-gnu-stage-final-gcc-debug-wrapper-10.3.0.drv': 3 dependencies couldn't be built
error: build of '/nix/store/f3qfckhryf4mnx9i0s7fca4bs221mlqa-riscv64-unknown-linux-gnu-stage-final-gcc-debug-wrapper-10.3.0.drv' failed
```

`riscv32` fails too:

```
these derivations will be built:
  /nix/store/3zp4hvr2zclf8gqz8scshfziklmk97a5-glibc-riscv32-unknown-linux-gnu-2.32-46.drv
  /nix/store/ch6jp6y75y3v9s4r7cj6vkl9dpbskh5a-riscv32-unknown-linux-gnu-binutils-wrapper-2.35.1.drv
  /nix/store/dynxn6gfjh11f805algbz84qxzjbkqiz-riscv32-unknown-linux-gnu-stage-final-gcc-debug-10.3.0.drv
  /nix/store/6qcbdsjb51m4z0mmkibk2l4iakx07nj4-riscv32-unknown-linux-gnu-stage-final-gcc-debug-wrapper-10.3.0.drv
building '/nix/store/3zp4hvr2zclf8gqz8scshfziklmk97a5-glibc-riscv32-unknown-linux-gnu-2.32-46.drv' on 'ssh://builder1'...
unpacking sources
unpacking source archive /nix/store/yl0acs9vzng0cd4qfdcgb3zpwjm034nw-glibc-2.32.tar.xz
source root is glibc-2.32
setting SOURCE_DATE_EPOCH to timestamp 1596593820 of file glibc-2.32/wctype/wctype_l.c
patching sources
applying patch /nix/store/gy22hz2hblz1lymvaqw752qpnmm44qsy-2.32-master.patch.gz
patching file sysdeps/x86_64/fpu/multiarch/ifunc-fma4.h
patching file NEWS
patching file iconv/Versions
patching file iconv/gconv_charset.c
patching file iconv/gconv_charset.h
patching file iconv/gconv_int.h
patching file iconv/iconv_open.c
patching file iconv/iconv_prog.c
patching file intl/dcigettext.c
patching file intl/tst-codeset.c
patching file NEWS
patching file stdio-common/Makefile
patching file stdio-common/errlist.c
patching file stdio-common/test-strerr.c
patching file sysdeps/gnu/errlist.h
patching file version.h
patching file sysdeps/aarch64/multiarch/memcpy_advsimd.S
patching file sysdeps/aarch64/multiarch/memcpy.c
patching file sysdeps/aarch64/multiarch/memmove.c
patching file sysdeps/unix/sysv/linux/aarch64/cpu-features.h
patching file NEWS
patching file sysdeps/unix/sysv/linux/Makefile
patching file sysdeps/unix/sysv/linux/semctl.c
patching file sysdeps/unix/sysv/linux/tst-sysvsem-linux.c
patching file sysvipc/test-sysvsem.c
patching file NEWS
patching file sysdeps/unix/sysv/linux/Makefile
patching file sysdeps/unix/sysv/linux/msgctl.c
patching file sysdeps/unix/sysv/linux/tst-sysvmsg-linux.c
patching file NEWS
patching file sysdeps/unix/sysv/linux/Makefile
patching file sysdeps/unix/sysv/linux/shmctl.c
patching file sysdeps/unix/sysv/linux/tst-sysvshm-linux.c
patching file manual/tunables.texi
patching file sysdeps/x86/cacheinfo.c
patching file sysdeps/x86/cacheinfo.c
patching file sysdeps/aarch64/dl-machine.h
patching file NEWS
patching file sysdeps/aarch64/start.S
patching file string/bits/string_fortified.h
patching file debug/Makefile
patching file debug/warning-nop.c
patching file misc/sys/cdefs.h
patching file support/Makefile
patching file support/resolv_response_context_duplicate.c
patching file support/resolv_response_context_free.c
patching file support/resolv_test.c
patching file support/resolv_test.h
patching file support/resolv_test.c
patching file support/resolv_test.h
patching file NEWS
patching file resolv/Makefile
patching file resolv/res_send.c
patching file resolv/tst-resolv-txnid-collision.c
patching file resolv/tst-resolv-txnid-collision.c
patching file sysdeps/generic/unwind.h
patching file NEWS
patching file sysdeps/sh/be/sh4/fpu/Implies
patching file sysdeps/sh/le/sh4/fpu/Implies
patching file sysdeps/unix/sysv/linux/sh/be/sh4/fpu/Implies
patching file sysdeps/unix/sysv/linux/sh/le/sh4/fpu/Implies
patching file NEWS
patching file iconv/tst-iconv_prog.sh
patching file iconvdata/ibm1364.c
patching file Rules
patching file malloc/Makefile
patching file iconvdata/Makefile
patching file iconvdata/bug-iconv13.c
patching file iconvdata/euc-kr.c
patching file iconvdata/ksc5601.h
patching file sysdeps/x86_64/multiarch/memmove-vec-unaligned-erms.S
patching file NEWS
patching file NEWS
patching file sysdeps/x86/Makefile
patching file sysdeps/x86/dl-cet.c
patching file sysdeps/x86/tst-setjmp-cet.c
patching file NEWS
patching file elf/Makefile
patching file elf/ifuncmain6pie.c
patching file elf/ifuncmod6.c
patching file sysdeps/i386/dl-machine.h
patching file sysdeps/x86_64/dl-machine.h
patching file sysdeps/aarch64/dl-bti.c
patching file sysdeps/aarch64/dl-bti.c
patching file elf/dl-load.c
patching file elf/dl-load.c
patching file elf/rtld.c
patching file sysdeps/aarch64/dl-prop.h
patching file sysdeps/generic/dl-prop.h
patching file sysdeps/generic/ldsodefs.h
patching file sysdeps/x86/dl-prop.h
patching file sysdeps/aarch64/dl-bti.c
patching file sysdeps/aarch64/dl-prop.h
patching file sysdeps/aarch64/linkmap.h
patching file stdio-common/vfscanf-internal.c
patching file sysdeps/aarch64/start.S
patching file iconvdata/Makefile
patching file iconvdata/bug-iconv14.c
patching file iconvdata/iso-2022-jp-3.c
patching file nscd/netgroupcache.c
patching file sysdeps/powerpc/powerpc64/backtrace.c
patching file sysdeps/s390/configure
patching file sysdeps/s390/configure.ac
patching file sysdeps/s390/memmove.c
patching file sysdeps/s390/multiarch/ifunc-impl-list.c
patching file stdlib/tst-secure-getenv.c
patching file support/capture_subprocess.h
patching file support/subprocess.h
patching file support/support_capture_subprocess.c
patching file support/support_subprocess.c
patching file elf/tst-env-setuid.c
patching file elf/Makefile
patching file elf/tst-env-setuid-tunables.c
patching file elf/dl-tunables.c
patching file elf/tst-env-setuid-tunables.c
patching file support/support_subprocess.c
patching file support/support_capture_subprocess.c
patching file support/support_subprocess.c
applying patch /nix/store/pp29isd0nvy4jrm9q2f92pjlc83xwa0i-nix-locale-archive.patch
patching file locale/loadarchive.c
patching file locale/programs/locale.c
Hunk #1 succeeded at 634 (offset 1 line).
Hunk #2 succeeded at 663 (offset 1 line).
patching file locale/programs/locarchive.c
applying patch /nix/store/5212825a2npxaf7qd81sxfrqq6219rlx-dont-use-system-ld-so-cache.patch
patching file elf/ldconfig.c
Hunk #1 succeeded at 55 with fuzz 1 (offset 4 lines).
patching file elf/Makefile
Hunk #1 succeeded at 647 (offset 88 lines).
patching file sysdeps/generic/dl-cache.h
applying patch /nix/store/mnglr8rr7nl444h7p50ysyq8qd0fm1lm-dont-use-system-ld-so-preload.patch
patching file elf/rtld.c
Hunk #1 succeeded at 1781 (offset 268 lines).
applying patch /nix/store/za0pg7fmysrcwrqcal26fnmzw6vycgdn-fix_path_attribute_in_getconf.patch
patching file sysdeps/unix/confstr.h
applying patch /nix/store/xkd50xxii6k7l1kmw4l5x6xzbhamcs87-allow-kernel-2.6.32.patch
patching file sysdeps/unix/sysv/linux/configure
patching file sysdeps/unix/sysv/linux/configure.ac
patching file sysdeps/unix/sysv/linux/dl-osinfo.h
applying patch /nix/store/3l70d7kcfsh91w6792h4fqs4kjbq17py-glibc-reinstate-prlimit64-fallback.patch?id=eab07e78b691ae7866267fc04d31c7c3ad6b0eeb
patching file sysdeps/unix/sysv/linux/getrlimit64.c
patching file sysdeps/unix/sysv/linux/setrlimit.c
Hunk #2 succeeded at 46 with fuzz 1.
patching file sysdeps/unix/sysv/linux/setrlimit64.c
Hunk #1 succeeded at 36 with fuzz 2.
applying patch /nix/store/nqa9d4yxz1l2cgswzqr1pkm4jfrksm0q-locale-C.diff
patching file localedata/locales/C
applying patch /nix/store/7kw224hdyxd7115lrqh9a4dv2x8msq2s-fix-x64-abi.patch
patching file elf/dl-tls.c
Hunk #1 succeeded at 898 (offset 71 lines).
updateAutotoolsGnuConfigScriptsPhase
Updating Autotools / GNU config script to a newer upstream version: ./scripts/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./scripts/config.guess
configuring
configure flags: --disable-static --prefix=/nix/store/j2p71vazickimddxak4lp4px2ycmjg8z-glibc-riscv32-unknown-linux-gnu-2.32-46 --bindir=/nix/store/2fq1f2wymkl3m5d35nfjqx6q9r74p6cz-glibc-riscv32-unknown-linux-gnu-2.32-46-bin/bin --sbindir=/nix/store/2fq1f2wymkl3m5d35nfjqx6q9r74p6cz-glibc-riscv32-unknown-linux-gnu-2.32-46-bin/sbin --includedir=/nix/store/n2a8hrl055hnhqwi7ww0yn0rafawygrb-glibc-riscv32-unknown-linux-gnu-2.32-46-dev/include --oldincludedir=/nix/store/n2a8hrl055hnhqwi7ww0yn0rafawygrb-glibc-riscv32-unknown-linux-gnu-2.32-46-dev/include --mandir=/nix/store/2fq1f2wymkl3m5d35nfjqx6q9r74p6cz-glibc-riscv32-unknown-linux-gnu-2.32-46-bin/share/man --infodir=/nix/store/2fq1f2wymkl3m5d35nfjqx6q9r74p6cz-glibc-riscv32-unknown-linux-gnu-2.32-46-bin/share/info --docdir=/nix/store/j2p71vazickimddxak4lp4px2ycmjg8z-glibc-riscv32-unknown-linux-gnu-2.32-46/share/doc/glibc --libdir=/nix/store/j2p71vazickimddxak4lp4px2ycmjg8z-glibc-riscv32-unknown-linux-gnu-2.32-46/lib --libexecdir=/nix/store/j2p71vazickimddxak4lp4px2ycmjg8z-glibc-riscv32-unknown-linux-gnu-2.32-46/libexec --localedir=/nix/store/j2p71vazickimddxak4lp4px2ycmjg8z-glibc-riscv32-unknown-linux-gnu-2.32-46/share/locale -C --enable-add-ons --sysconfdir=/etc --enable-stackguard-randomization --with-headers=/nix/store/7v9w44d5pyqydq7mhgywz5sqkhqarcak-linux-headers-5.12/include --disable-profile --enable-kernel=3.2.0 --without-fp --with-__thread --build=x86_64-unknown-linux-gnu --host=riscv32-unknown-linux-gnu
configure: loading cache config.cache
checking build system type... x86_64-unknown-linux-gnu
checking host system type... riscv32-unknown-linux-gnu
checking for riscv32-unknown-linux-gnu-gcc... riscv32-unknown-linux-gnu-gcc
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether riscv32-unknown-linux-gnu-gcc accepts -g... yes
checking for gcc... gcc
checking for riscv32-unknown-linux-gnu-readelf... riscv32-unknown-linux-gnu-readelf
checking whether we are using the GNU C++ compiler... no
checking whether riscv32-unknown-linux-gnu-g++ accepts -g... no
checking whether riscv32-unknown-linux-gnu-g++ can link programs... no
checking for sysdeps preconfigure fragments... aarch64 alpha arc arm csky hppa i386 m68k microblaze checking for grep that handles long lines and -e... /nix/store/7n3yzh9wza4bdqc04v01xddnfhkrwk2a-gnugrep-3.6/bin/grep
checking for egrep... /nix/store/7n3yzh9wza4bdqc04v01xddnfhkrwk2a-gnugrep-3.6/bin/grep -E
mips nios2 powerpc riscv glibc does not yet support 32-bit systems
builder for '/nix/store/3zp4hvr2zclf8gqz8scshfziklmk97a5-glibc-riscv32-unknown-linux-gnu-2.32-46.drv' failed with exit code 1
error: build of '/nix/store/3zp4hvr2zclf8gqz8scshfziklmk97a5-glibc-riscv32-unknown-linux-gnu-2.32-46.drv' on 'ssh://builder1' failed: builder for '/nix/store/3zp4hvr2zclf8gqz8scshfziklmk97a5-glibc-riscv32-unknown-linux-gnu-2.32-46.drv' failed with exit code 1
builder for '/nix/store/3zp4hvr2zclf8gqz8scshfziklmk97a5-glibc-riscv32-unknown-linux-gnu-2.32-46.drv' failed with exit code 1
cannot build derivation '/nix/store/ch6jp6y75y3v9s4r7cj6vkl9dpbskh5a-riscv32-unknown-linux-gnu-binutils-wrapper-2.35.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/dynxn6gfjh11f805algbz84qxzjbkqiz-riscv32-unknown-linux-gnu-stage-final-gcc-debug-10.3.0.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/6qcbdsjb51m4z0mmkibk2l4iakx07nj4-riscv32-unknown-linux-gnu-stage-final-gcc-debug-wrapper-10.3.0.drv': 3 dependencies couldn't be built
error: build of '/nix/store/6qcbdsjb51m4z0mmkibk2l4iakx07nj4-riscv32-unknown-linux-gnu-stage-final-gcc-debug-wrapper-10.3.0.drv' failed
```

Also it works on my main system (21.05 stable I think?) so it broke not long ago. I'll try to bisect it.

**Comment by piegamesde** [comment]
`git bisect` result tells that the package broke in 166948d4794be5c56c6279bf580ac416b1bcf022 from #123989 by @Mic92

**Comment by Mic92** [comment]
I think this is solved by https://github.com/NixOS/nixpkgs/pull/129004

**Comment by rowanG077** [comment]
@Mic92 My build was from the master branch of nixpkgs yesterday. That PR was already merged so It didn't solve it.

**Comment by piegamesde** [comment]
@rowanG077 No, it was merged against `staging` and has not landed on master yet.

**Comment by rowanG077** [comment]
@piegamesde Ah yes I missed that sorry. I'm trying out current staging to see if it is fixed.

**Comment by piegamesde** [comment]
It works
